pkg/cmd/pulumi/stack.go

```diff
@@ -233,11 +234,21 @@ func stringifyOutput(v interface{}) string {
 		return s
 	}
 
-	b, err := json.Marshal(v)
-	if err != nil {
+	// json.Marshal escapes HTML characters, which we don't want,
```

Comment:
> The only way to control this in Go is with json.Encoder.
However, that unconditionally places a newline at the end,
so we have to strip that for our use case.

I really hate this, I've ran into this wall multiple times and it's just infuriating that they don't expose an option here 😭 

There's probably some other places in the codebase doing this trick (or a poor attempt at it). Probably worth scanning and seeing if it makes sense to make a util helper function to do this `json.MarshalUnescaped`?

Reply:
I considered that based on your comment in #10561, but the only place that this actually matters is at the presentation layers: where the JSON is surfaced to the user in a view somewhere.
For machines parsing the JSON, the escaped HTML is still a valid JSON string representing the same information, and the extraneous newline is not a problem.

Reply:
> but the only place that this actually matters is at the presentation layers

Mostly, but there's a few places where we need to do this in the presentation layers. 

There is in fact already a helper function `makeJSONString` (in pkg/cmd/pulumi/util.go) that's basically doing the same as here except without the newline handling. 

Should probably just change this to use that, and fix that to handle the newline.

Reply:
Alright, I'll try to merge this into makeJSONString.

Reply:
@Frassle I merged them, but makeJSONString was previously only for indented output. 
In indented output form, it should not strip the trailing newline—because the output is likely going to be printed on the console (e.g. it's used by `stack output --json`).

So I've added a multiline flag that decides whether to indent or not, and also whether to strip the trailing newline.
EscapeHTML is always disabled.